sound/soc/sof/intel/hda-dsp.c

```diff
@@ -237,6 +237,25 @@ bool hda_dsp_core_is_enabled(struct snd_sof_dev *sdev,
 	return is_enable;
 }
 
+bool hda_dsp_core_is_disabled(struct snd_sof_dev *sdev,
+			      unsigned int core_mask)
+{
+	int val;
+	bool is_disabled;
+
+	val = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_REG_ADSPCS);
+
+	is_disabled = !(val & HDA_DSP_ADSPCS_CPA_MASK(core_mask)) &&
+		      !(val & HDA_DSP_ADSPCS_SPA_MASK(core_mask)) &&
```

Comment:
why do we need to check SPA? Isnt CPA enough?

Reply:
Im still confused about what issue this fixes? Where do you see this problem where you expect the core to be OFF but it is in the middle of a transition?